src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/CreateCollectionCommand.java

```diff
@@ -77,10 +73,90 @@ public record VectorSearchConfig(
         this.dimension = dimension;
         this.metric = metric == null ? "cosine" : metric;
+        this.vectorizeConfig = vectorizeConfig;
+      }
+
+      public record VectorizeConfig(
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service provider",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("provider")
+              String provider,
+          @NotNull
+              @Schema(
+                  description = "Registered Embedding service model",
+                  type = SchemaType.STRING,
+                  implementation = String.class)
+              @JsonProperty("model_name")
+              String modelName,
+          @Valid
+              @NotNull
+              @Schema(
+                  description = "Authentication config for chosen embedding service",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceAuthentication.class)
+              @JsonProperty("authentication")
+              VectorizeServiceAuthentication vectorizeServiceAuthentication,
+          @Valid
+              @JsonInclude(JsonInclude.Include.NON_NULL)
+              @Nullable
+              @Schema(
+                  description =
+                      "Optional parameters that match the template provided for the provider",
+                  type = SchemaType.OBJECT,
+                  implementation = VectorizeServiceParameter.class)
+              @JsonProperty("parameters")
+              VectorizeServiceParameter vectorizeServiceParameter) {
+        public record VectorizeServiceAuthentication(
+            @NotNull
+                @Schema(
+                    description =
+                        "List of authentications that can be used when sending documents that need vectorization. One or more of \"NONE\", \"HEADER\", \"SHARED_SECRET\"",
+                    type = SchemaType.ARRAY,
+                    implementation = String.class)
+                @JsonProperty("type")
+                List<
+                        @Pattern(
+                            regexp = "(NONE|HEADER|SHARED_SECRET)",
+                            message =
+                                "authentication type can only be one or more of 'NONE', 'HEADER' or 'SHARED_SECRET'")
+                        String>
+                    type,
+            @JsonInclude(JsonInclude.Include.NON_NULL)
+                @Nullable
+                @Schema(
+                    description =
+                        "Secret name. when stored_secrets authentication is used must be provided with the name of a pre-registered secret",
+                    type = SchemaType.STRING,
+                    implementation = String.class)
+                @JsonProperty("secret_name")
+                String secretName) {}
+
+        public record VectorizeServiceParameter(
```

Comment:
It's better to resolve parameters as a map and validate required parameter are available based on the provider. Predefined fields here will be difficult to explain on the schema. Let's discuss this out.

Reply:
changed into map